src/data/nat/prime.lean

```diff
@@ -408,6 +408,20 @@ begin
       rw e, exact pow_dvd_pow _ l } }
 end
 
+/--
+If `p` is prime,
+and `a` doesn't divide `p^k`, but `a` does divide `p^(k+1)`
+then `a = p^(k+1)`.
+-/
+lemma eq_prime_pow_of_dvd_least_prime_pow
+  {a p k : ℕ} (pp : prime p) (h₁ : ¬(a ∣ p^k)) (h₂ : a ∣ p^(k+1)) :
+  a = p^(k+1) :=
+begin
+  rcases (dvd_prime_pow pp).1 h₂ with ⟨l, ⟨h, rfl⟩⟩,
```

Comment:
Nowadays the cool way to write this is
```
   obtain ⟨l, ⟨h, rfl⟩⟩ := (dvd_prime_pow pp).1 h₂,
```
and the nested `⟨⟩`s should also be unnecessary.

Reply:
Done. I should add `obtain` to my regular arsenal! I've left the nested `⟨⟩`; perhaps they help decipherability after the fact?

Reply:
Small note: the cool thing about `obtain` is that you can add a type ascription to increase readability:
```lean
obtain ⟨l, ⟨h, rfl⟩⟩ : \exists l, blabla := (dvd_prime_pow pp).1 h₂,
```